rs/ethereum/cketh/minter/src/eth_logs/parser.rs

```diff
@@ -164,22 +164,29 @@ impl LogParser for ReceivedEthOrErc20LogParser {
 fn ensure_not_pending(entry: &LogEntry) -> Result<(BlockNumber, EventSource), ReceivedEventError> {
     let _block_hash = entry
         .block_hash
+        .clone()
```

Comment:
I don't think cloning is necessary here:
```suggestion
    let _block_hash = entry
        .block_hash
        .as_ref()
        .ok_or(ReceivedEventError::PendingLogEntry)?;
```

Reply:
Good catch, thanks. This could be done for `_transaction_index` as well. See https://github.com/dfinity/ic/pull/5428/commits/0eb8d5cb8a09977b582d83554e3a939a2533b4a5